tutorials/histfactory/hf001_example.C

```diff
@@ -89,7 +90,103 @@ void hf001_example() {
   // meas.PrintXML( "xmlFromCCode", meas.GetOutputFilePrefix() );
 
   // Now, do the measurement
-  MakeModelAndMeasurementFast( meas );
-
-
+  std::unique_ptr<RooWorkspace> ws{MakeModelAndMeasurementFast(meas)};
+
+  RooStats::ModelConfig *modelConfig = static_cast<RooStats::ModelConfig *>(ws->obj("ModelConfig"));
+
+  // Get probability density function and parameters list from model
+  RooAbsPdf *pdf = modelConfig->GetPdf();
+  RooArgSet globalObservables{*modelConfig->GetGlobalObservables()};
+
+  /*
+  parameters in globalObservables:
+
+  nominalLumi
+  nom_alpha_syst1 -> was set as constant in the beginning
+  nom_alpha_syst2
+  nom_alpha_syst3
+  nom_gamma_stat_channel1_bin_0
+  nom_gamma_stat_channel1_bin_1
+
+  */
+
+  // Perform the fit
+  using namespace RooFit;
+  std::unique_ptr<RooFitResult> result{
+     pdf->fitTo(*ws->data("obsData"), Save(), PrintLevel(-1), GlobalObservables(globalObservables))};
+
+   // Getting list of Parameters of Interest and getting first from them
+   RooRealVar* poi = static_cast<RooRealVar*>(modelConfig->GetParametersOfInterest()->first());
+
+   std::unique_ptr<RooAbsReal> nll{pdf->createNLL(*ws->data("obsData"))};
+   std::unique_ptr<RooAbsReal> profile{nll->createProfile(*poi)};
+
+   // frame for future plot
+   std::unique_ptr<RooPlot> frame{poi->frame()};
+
+   frame->SetTitle("");
+   frame->GetYaxis()->SetTitle("-log likelihood");
+   frame->GetXaxis()->SetTitle(poi->GetTitle());
+
+   TCanvas profileLikelihoodCanvas{"combined", "",800,600};
+   
+   double xmin = poi->getMin();
+   double xmax = poi->getMax();
+   TLine * line = new TLine(xmin,.5,xmax,.5);
+   line->SetLineColor(kGreen);
+   TLine * line90 = new TLine(xmin,2.71/2.,xmax,2.71/2.);
+   line90->SetLineColor(kGreen);
+   TLine * line95 = new TLine(xmin,3.84/2.,xmax,3.84/2.);
+   line95->SetLineColor(kGreen);
+   frame->addObject(line);
+   frame->addObject(line90);
+   frame->addObject(line95);
+   
+   nll->plotOn(frame.get(), ShiftToZero(), LineColor(kRed), LineStyle(kDashed));
+   profile->plotOn(frame.get());
+
+   frame->SetMinimum(0);
+   frame->SetMaximum(2.);
+
+   frame->Draw();
+
+   // Save drawed picture as PNG file
+   std::string profilePlotName = "LikelihoodCurve.png";
+   profileLikelihoodCanvas.SaveAs( profilePlotName.c_str() );
```

Comment:
```suggestion
```
I wouldn't do this in this C++ tutorial. The plot is already shown on the sceen, and people already know how to save a TCanvas.